File: 0226-invert-binary-tree/0226-invert-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:

        def swap(root):
            if not root:
                return

            root.left, root.right = root.right, root.left

            swap(root.left)
            swap(root.right)

        swap(root)

        return root

File: 0226-invert-binary-tree/0226-invert-binary-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func invertTree(root *TreeNode) *TreeNode {
    var invert func(root *TreeNode)

    invert = func(root *TreeNode) {

        if (root == nil) || 
        ( root.Left == nil && root.Right == nil) {
            return
        }

        temp := root.Left
        root.Left = root.Right
        root.Right = temp
        invert(root.Left)
        invert(root.Right)
    }

    invert(root)
    return root
}